core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -313,6 +313,16 @@ private SqlLibraryOperators() {
           OperandTypes.INTEGER,
           SqlFunctionCategory.STRING);
 
+  @LibraryOperator(libraries = {MYSQL, POSTGRESQL, ORACLE})
+  public static final SqlFunction REGEXP_REPLACE =
+      new SqlFunction("REGEXP_REPLACE",
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.cascade(ReturnTypes.explicit(SqlTypeName.VARCHAR),
+              SqlTypeTransforms.TO_NULLABLE),
```

Comment:
I see that MYSQL and Oracle support another 3 optional arguments for `REGEXP_REPLACE`[1], are we planning to support this ?

[1] https://dev.mysql.com/doc/refman/8.0/en/regexp.html#function_regexp-replace

Reply:
Hi @danny0405 , Thanks for your replies. I have review [other databases'](https://issues.apache.org/jira/browse/CALCITE-3280?focusedCommentId=16913081&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16913081) usage about `regexp_replace`. Maybe there is a final answer about this question.

Do you think we should support this? I can work for it If it's needed.

Reply:
It would be nice if we support a MYSQL style `REGEXP_REPLACE`.

Reply:
OK, I will update it later.

Reply:
I updated the code as you said by following the `Mysql`'s style.